exonum/src/node/mod.rs

```diff
@@ -760,10 +762,28 @@ impl fmt::Debug for ApiSender {
     }
 }
 
+fn deserialize_socket_address<'de, D>(value: D) -> Result<SocketAddr, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let address_str: String = Deserialize::deserialize(value)?;
+    address_str
+        .to_socket_addrs()
+        .map_err(de::Error::custom)?
+        .next()
+        .ok_or_else(|| {
+            de::Error::custom(&format!(
+                "no one ip belongs to the hostname: {}",
+                address_str
+            ))
+        })
+}
```

Comment:
I'd say that hostname support should be more involved.

Usually when `getaddrinfo()` (deep within `to_socket_addrs()` implementation) returns more than one address then the application is expected to try them out until one of them connects. So in my opinion it would be better to do

```diff
 pub struct ConnectInfo {
      /// Peer address.
-     pub address: SocketAddr,
+     pub address: String,
```

and then handle hostnames in the connection code.

Reply:
I thought about this and even have tried to do so. But such approach depends on `Connect` message wich plans to remove in future. Therefor for now we've decided to use simple approach. More intellegent and complex version will be done in another ticket.